remote/bitbucketserver/bitbucketserver.go

```diff
@@ -0,0 +1,358 @@
+
+	client := NewClientWithToken(bs.ConsumerKey, bs.URL, u.Token)
+	fileURL := fmt.Sprintf("%s/projects/%s/repos/%s/browse/%s?raw", bs.URL,r.Owner,r.Name,f)
+	log.Info(fileURL)
+	response, err := client.Get(fileURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if response.StatusCode == 404 {
+		return nil,nil
+	}
+	defer response.Body.Close()
+	responseBytes, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+
+	return  responseBytes, nil;
+}
+
+func (bs *BitbucketServer) Status(u *model.User, r *model.Repo, b *model.Build, link string) error{
+	log.Info("Staring status for bitbucketServer")
+	return nil;
+}
+
+func (bs *BitbucketServer) Netrc(user *model.User, r *model.Repo) (*model.Netrc, error){
+	log.Info("Starting the Netrc lookup")
+	u, err := url.Parse(bs.URL)
+	if err != nil {
+		panic(err)
+	}
+	return &model.Netrc{
+		Machine:  u.Host,
+		Login:    bs.GitUserName,
+		Password: bs.GitPassword,
+	}, nil
+}
+
+func (bs *BitbucketServer) Activate(u *model.User, r *model.Repo, k *model.Key, link string) error{
+	log.Info(fmt.Sprintf("Staring activate for bitbucketServer user: %s repo: %s key: %s link: %s",u.Login,r.Name,k,link))
+	client := NewClientWithToken(bs.ConsumerKey, bs.URL, u.Token)
+	hook, err := bs.CreateHook(client, r.Owner,r.Name, "com.atlassian.stash.plugin.stash-web-post-receive-hooks-plugin:postReceiveHook",link)
+	if err !=nil {
+		return err
+	}
+	log.Info(hook)
+	return nil;
+}
+
+func (bs *BitbucketServer) Deactivate(u *model.User, r *model.Repo, link string) error{
+	log.Info(fmt.Sprintf("Staring deactivating for bitbucketServer user: %s repo: %s link: %s",u.Login,r.Name,link))
+	client := NewClientWithToken(bs.ConsumerKey, bs.URL, u.Token)
+	err := bs.DeleteHook(client, r.Owner,r.Name, "com.atlassian.stash.plugin.stash-web-post-receive-hooks-plugin:postReceiveHook",link)
+	if err !=nil {
+		return err
+	}
+	return nil;
+}
+
+func (bs *BitbucketServer) Hook(r *http.Request) (*model.Repo, *model.Build, error){
+	log.Info("Staring hook for bitbucketServer")
+	defer r.Body.Close()
+	contents, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Info(err)
+	}
+
+	var hookPost postHook
+	json.Unmarshal(contents, &hookPost)
+
+
+
+	buildModel := &model.Build{}
+	buildModel.Event = model.EventPush
+	buildModel.Ref = hookPost.RefChanges[0].RefID
+	buildModel.Author = hookPost.Changesets.Values[0].ToCommit.Author.EmailAddress
+	buildModel.Commit = hookPost.RefChanges[0].ToHash
+	buildModel.Avatar = avatarLink(hookPost.Changesets.Values[0].ToCommit.Author.EmailAddress)
+
+	//All you really need is the name and owner. That's what creates the lookup key, so it needs to match the repo info. Just an FYI
+	repo := &model.Repo{}
+	repo.Name=hookPost.Repository.Slug
+	repo.Owner = hookPost.Repository.Project.Key
+	repo.AllowTag=false
+	repo.AllowDeploy=false
+	repo.AllowPull=false
+	repo.AllowPush=true
```

Comment:
no need to set these values to false because the zero value of a boolean in Go is false. So you can remove these lines :)